docs/man-openmpi/man3/MPI_Info_create_env.3.rst

```diff
@@ -0,0 +1,80 @@
+.. _mpi_info_create_env:
+
+
+MPI_Info_create_env
+===================
+
+.. include_body
+
+:ref:`MPI_Info_create_env` - Creates a new info object with the same construction as MPI_INFO_ENV as created during :ref:`MPI_Init`  or :ref:`MPI_Init_thread` when the same arguments
+are used.
+
+
+SYNTAX
+------
+
+
+C Syntax
+^^^^^^^^
+
+.. code-block:: c
+
+   #include <mpi.h>
+
+   int MPI_Info_create_env(int argc, char *argv[], MPI_Info *info)
+
+
+Fortran Syntax
+^^^^^^^^^^^^^^
+
+.. code-block:: fortran
+
+   USE MPI
+   ! or the older form: INCLUDE 'mpif.h'
+   MPI_INFO_CREATE_ENV(INFO, IERROR)
+   	INTEGER	INFO, IERROR
+
+
+Fortran 2008 Syntax
+^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: fortran
+
+   USE mpi_f08
+   MPI_Info_create_env(info, ierror)
+   	TYPE(MPI_Info), INTENT(OUT) :: info
+   	INTEGER, OPTIONAL, INTENT(OUT) :: ierror
+
+
+OUTPUT PARAMETERS
+-----------------
+* ``info``: Info object created (handle).
+* ``IERROR``: Fortran only: Error status (integer).
+
+DESCRIPTION
+-----------
+
+:ref:`MPI_Info_create_env` creates a new info object and populates it with the same key/value pairs as are present in MPI_INFO_ENV.
+
+Note
+----
+
+:ref:`MPI_Info_create_env` is one of the few functions that can be called
+before :ref:`MPI_Init` and after :ref:`MPI_Finalize`.
+
+ERRORS
+------
+
+Almost all MPI routines return an error value; C routines as the value
+of the function and Fortran routines in the last argument.
+
+Before the error value is returned, the current MPI error handler is
+called. By default, this error handler aborts the MPI job, except for
+I/O function errors. The error handler may be changed with
+:ref:`MPI_Comm_set_errhandler`; the predefined error handler MPI_ERRORS_RETURN
```

Comment:
Is this really true?  If so, on which communicator do you set the errhandler to affect exceptions on MPI APIs that do not have communicator arguments?

E.g., what if this function generates an exception in the Sessions model (i.e., if MPI_INIT is never invoked, and we therefore can't set an errhandler on `COMM_WORLD` or `COMM_SELF`)?

I honestly don't remember where these MPI-4 discussions landed, and I just spent 10 minutes looking through the MPI-4 doc and didn't find the answer.  ☹️

Reply:
A possible way around it is for MPI-4.1 to allow for setting error handlers on `MPI_SESSION_NULL` before `MPI_Init` or `MPI_Session_init`, with fatal (or abort) error handler as default. 

For the tame being, under the principle that errors should not pass silently (do we all agree on this?) and the fact that MPI users are quite lazy to consistently check for errors, perhaps the best is to abort.

Reply:
There was always the argument in the Forum that the `*_NULL` handles are *invalid*, and therefore we can't actually do anything with them except to test for them as sentinel values.

I am 100% certain that there was discussion for how to set / get error handlers for the functions that can be called before session_init (i.e., functions that do not have an errhandler-carrying MPI handle as an argument -- and therefore have no errhandler to invoke; this role used to be filled by using the errhandlers from `MPI_COMM_SELF`, but in the Sessions model, `MPI_COMM_SELF` has not been constructed/initialized, and therefore isn't relevant); I just don't remember where it landed / what specifically was decided.

Reply:
see section 9.3 of the MPI 4.0 standard for discussion about this topic.

Reply:
> There was always the argument in the Forum that the *_NULL handles are invalid

Do you mean like `MPI_FILE_NULL` :disappointed: ?

Reply:
> see section 9.3 of the MPI 4.0 standard for discussion about this topic.

Ah here it is -- I couldn't find this last night: MPI-4.0 section 10.3 page 459:6-11:

> MPI calls that are not related to any MPI objects are considered to be attached to the communicator MPI_COMM_SELF when using the World Model (see Section 11.2).  When MPI_COMM_SELF is not initialized (i.e., before MPI_INIT / MPI_INIT_THREAD, after MPI_FINALIZE, or when using the Sessions Model exclusively) the error raises the initial error handler (set during the launch operation, see 11.8.4).

Section 11.8.4 refers to the `mpi_initial_errhandler` MPI_Info key to `MPI_COMM_SPAWN[_MULTIPLE]`.  Section 11.2 p490:16 also states that `mpi_initial_errhandler` is a CLI argument to `mpiexec`.

@hppritcha So that's what this text should state -- that you use `MPI_COMM_SET_ERRHANDLER` on `MPI_COMM_SELF` when using the World model, or the `mpi_initial_errhandler` CLI argument to `mpiexec` or info key to `MPI_COMM_SPAWN[_MULTIPLE]`.

> Do you mean like `MPI_FILE_NULL` 😞 ?

I assume you are referring to MPI-4.0 p720:10-12:

> For I/O errors that are not associated with a valid file handle (e.g., in MPI_FILE_OPEN or MPI_FILE_DELETE), the first argument passed to the error handler is MPI_FILE_NULL.

(and the whole paragraph that comes after that, talking about special handling for `MPI_FILE_NULL`)

@dalcinl I'm not going to claim that MPI is 100% consistent with itself. ☹️  There at least is a recognition of this, and a Rationale block on line 24 on that page that explains *why*.

Reply:
@hppritcha Agreed.  I'll bet it's not in the OMPI schitzo component -- we need to add it to the list.

Reply:
ah it looks like that support is in the prte ompi schizo component.